packages/material-ui-lab/src/useAutocomplete/useAutocomplete.js

```diff
@@ -390,11 +390,47 @@ export default function useAutocomplete(props) {
         }
       }
     }
-  };
+  });
 
   React.useEffect(() => {
```

Comment:
All of this looks like derived state. Do we need an effect here or can we update the state during render instead?

Reply:
I suspect it needs to be in an effect to be able to set the scroll and `data-focus` attribute on the right option.

Reply:
Adding it to the backlog. We should split derived state and effects.